core/src/main/scala/org/apache/spark/metrics/MetricsConfig.scala

```diff
@@ -47,21 +47,27 @@ private[spark] class MetricsConfig(val configFile: Option[String]) extends Loggi
     setDefaultProperties(properties)
 
     // If spark.metrics.conf is not set, try to get file in class path
-    var is: InputStream = null
-    try {
-      is = configFile match {
-        case Some(f) => new FileInputStream(f)
-        case None => Utils.getSparkClassLoader.getResourceAsStream(METRICS_CONF)
+    (configFile match {
```

Comment:
cool, thanks, went with the first one.

incidentally I had a `Some()` where I should have had an `Option()`, which your candidates all fix, that I think was breaking the tests